### PR TITLE
Fix type hints

### LINKED ISSUE #109: Type hints of __init__ in StaticItem are not correct
The `__init__` function of [StaticItem](https://github.com/openzim/python-scraperlib/blob/4f8c3cc0ab5ffd3f2ea9c52937b86a5ec4c42bf6/src/zimscraperlib/zim/items.py#L48) is inherited from [Item](https://github.com/openzim/python-scraperlib/blob/4f8c3cc0ab5ffd3f2ea9c52937b86a5ec4c42bf6/src/zimscraperlib/zim/items.py#L26).

The type hint is incorrect in `Item` because it miss at least the `callback` attribute.

The type hint is even more incorrect in `StaticItem` because it misses all additional attributes from this class.

In addition to these two issues, I have a **newbie** question. Can't we write the `__init__` in a less generic way, i.e. with all attributes specified instead of `**kwargs` ? I imagine this would imply more maintenance every time `libzim` is modified, but at the same time would help a lot to discover all possible attributes + have correct type hints for every of them. But I might probably miss something else.

### LINKED ISSUE #112: Type hints of `convert_image` is not correct.
For now the type hints of `convert_image` is:
```python
def convert_image(
    src: pathlib.Path, dst: pathlib.Path, **params: Optional[dict]
) -> None:
```
But we can use `io.BytesIO()` as src and dst. 
Also, When I try to put the `fmt="PNG"` as a parameter, the Pyright will complain : `Type "Literal['PNG']" cannot be assigned to type "dict[Unknown, Unknown] | None`

### PR DESCRIPTION
**Pull request by benoit74**
Fix #109
Fix #112 

## Rationale

Type hints were too restrictive / incorrect

## Changes

- `Item`
  - for `**kwargs`, the type specified must be the one of individual param values, not the whole dictionary 
  - the `Union[str, bool, bytes]` type hint was replaced by `Any`, because the goal of this param is precisely to pass any type which could be further used
  - `FileItem` and `URLItem` were modified as well because they were very close to `StaticItem`
  - no change was necessary in `StaticItem` in fact
  - references to callback in docstring has been removed because it has been moved to the `add_item` method quite few years ago
- `convert_image`
  - `src` and `dst` can be either a `Pathlib.Path` or an `io.BytesIO` (maybe even `typing.BinaryIO` or `typing.IO[bytes]`, but I did not find a convenient way to test this to confirm that adherence to `typing.IO[bytes]` contract is sufficient or if we need something additional from `io.BytesIO`)
  - for `**params`, the type specified must be the one of individual param values, not the whole dictionary ; based on existing codebase across all scrapers I assumed that `str` is sufficient.
  - `save_image` had to be changed as well to reflect changes made in `convert_image`

### CODE REVIEW

File: src/zimscraperlib/zim/items.py
Comment (rgaudin) [inline src/zimscraperlib/zim/items.py:107]: I think that all *expected* (for which we have processing in the class) values should be individual optionnal keyword arguments with they expected type. We can leave `Any` for the rest

Comment (rgaudin) [reply src/zimscraperlib/zim/items.py:107]: Same logic would apply to classes inheriting `Item`

**Review by rgaudin — CHANGES_REQUESTED** [review]
Looks good ; please see my comments regarding kwargs

File: src/zimscraperlib/zim/items.py
Comment (benoit74) [reply src/zimscraperlib/zim/items.py:107]: I can only agree :)

File: src/zimscraperlib/image/transformation.py
Comment (benoit74) [reply src/zimscraperlib/image/transformation.py:22]: Yes, correct. This is an assumption, but it holds true on what we use so far (and what I found on the web)

**Comment by benoit74** [comment]
Changes done

Nota: please do not merge, I will rebase/push-force before merge first since another PR has been merged in the mean time.

**Review by rgaudin — COMMENTED** [review]
Looks good to me but we're now storing several variables per object for stuff that might never be set.
Given those `Item` are created in very huge number and only released at an unknown/uncontrolled point in the future (when libzim is done copying to disk) I wonder if it's a good idea.

**Comment by benoit74** [comment]
I don't know ... you are right, but somehow this is the price to pay to have proper typing in the whole class, otherwise we will still rely on `get_attr` and mostly untyped values.

Should we give it a try or rollback the change?

**Comment by benoit74** [comment]
By rollbacking the change I mean changing only the method signatures, it is already a step forward.

**Comment by rgaudin** [comment]
Yes the method signature is what's most important because that's the public API.

Could you use the `getattr` way for the optional stuff in an independent commit so we have a better view at both options?

**Comment by benoit74** [comment]
I pushed one more commit to "rollback" what is necessary to keep the get_attr approach while still enhancing the public API.

I tend to admit I prefer this "middle-ground" approach.

**Comment by benoit74** [comment]
Nota: last push is just a rebase on main last commit.

**Pull request merged**
